src/GeniusYield/Providers/Common.hs

```diff
@@ -119,6 +121,20 @@ fromJson b = do
   x <- first DeserializeErrorScriptDataJson $ Api.scriptDataFromJson Api.ScriptDataJsonDetailedSchema v
   pure . fromJust . fromData $ Api.toPlutusData $ Api.getScriptData x
 
+makeLastEraEndUnbounded :: Api.EraHistory -> Api.EraHistory
+makeLastEraEndUnbounded eh =
+  let Ouroboros.Summary eraList = extractEraSummaries eh
+   in Api.EraHistory $ Ouroboros.mkInterpreter $ Ouroboros.Summary $ g eraList
+ where
+  g eraList =
+    let eraList' = nonEmptyToList eraList
+        f [] = []
+        f [x] =
+          let oldEraParams = Ouroboros.eraParams x
+           in [x {Ouroboros.eraEnd = Ouroboros.EraUnbounded, Ouroboros.eraParams = oldEraParams {Ouroboros.eraSafeZone = Ouroboros.UnsafeIndefiniteSafeZone}}]
```

Comment:
Utilising `UnsafeIndefiniteSafeZone` is perhaps not required as when I do `IOG.epochInfoSlotToRelativeTime_ ei (slotToApi $ slotFromWord64 5000000)` without it inside privnet, I am able to get successful result.